src/test/java/tools/jackson/databind/jsontype/TestTypedSerializationWithSealedTypes.java

```diff
@@ -0,0 +1,216 @@
+        Map<String,Object> result = writeAndMap(m, new AnimalWrapper(new Dog("Amadeus", 7)));
+        // First level, wrapper
+        assertEquals(1, result.size());
+        List<?> l = (List<?>) result.get("animal");
+        assertNotNull(l);
+        assertEquals(2, l.size());
+        assertEquals(Dog.class.getName(), l.get(0));
+        Map<?,?> doggie = (Map<?,?>) l.get(1);
+        assertNotNull(doggie);
+        assertEquals(2, doggie.size());
+        assertEquals("Amadeus", doggie.get("name"));
+        assertEquals(Integer.valueOf(7), doggie.get("boneCount"));
+    }
+
+    /* !!! 30-Jan-2010, tatus: I am not completely sure below works as it should
+     *    Problem is, context of "untyped" map should prevent type information
+     *    being added to Animal entries, because Object.class has no type.
+     *    If type information is included, it will not be useful for deserialization,
+     *    since static type does not carry through (unlike in serialization).
+     *
+     *    But it is not quite clear how type information should be pushed through
+     *    array types...
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testInArray() throws Exception
+    {
+        // ensure we'll use mapper with default configs
+        ObjectMapper m = jsonMapperBuilder()
+                // ... so this should NOT be needed...
+                .deactivateDefaultTyping()
+                .build();
+
+        Animal[] animals = new Animal[] { new Cat("Miuku", "white"), new Dog("Murre", 9) };
+        Map<String,Object> map = new HashMap<String,Object>();
+        map.put("a", animals);
+        String json = m.writeValueAsString(map);
+        Map<String,Object> result = m.readValue(json, Map.class);
+        assertEquals(1, result.size());
+        Object ob = result.get("a");
+        if (!(ob instanceof List<?>)) {
+            // 03-Feb-2010, tatu: Weird; seems to fail sometimes...
+            fail("Did not map to entry with 'a' as List (but as "+ob.getClass().getName()+"): JSON == '"+json+"'");
+        }
+        List<?> l = (List<?>)ob;
+        assertNotNull(l);
+        assertEquals(2, l.size());
+        Map<?,?> a1 = (Map<?,?>) l.get(0);
+        assertEquals(3, a1.size());
+        String classProp = Id.CLASS.getDefaultPropertyName();
+        assertEquals(Cat.class.getName(), a1.get(classProp));
+        Map<?,?> a2 = (Map<?,?>) l.get(1);
+        assertEquals(3, a2.size());
+        assertEquals(Dog.class.getName(), a2.get(classProp));
+    }
+
+    /**
+     * Simple unit test to verify that serializing "empty" beans is ok
+     */
+    @Test
+    public void testEmptyBean() throws Exception
+    {
+        ObjectMapper m = jsonMapperBuilder()
+                .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false)
+                .build();
+        assertEquals("{\"@type\":\"empty\"}", m.writeValueAsString(new Empty()));
+    }
+
+    @Test
```

Comment:
Not even sure what it's trying to test, let's remove (I realize this is a copy of existing test, but no need to copy).